api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -1284,6 +1289,42 @@ public Thread run() {
         }
     }
 
+    class CleanableWorkerThreadFactory implements ThreadFactory {
+
+        private final ThreadGroup threadGroup;
+        private final long stackSize;
+        private final boolean markThreadAsDaemon;
+
+        CleanableWorkerThreadFactory(final ThreadGroup threadGroup, final long stackSize, final boolean markThreadAsDaemon) {
+            this.threadGroup = threadGroup;
+            this.stackSize = stackSize;
+            this.markThreadAsDaemon = markThreadAsDaemon;
+        }
+
+        public Thread newThread(final Runnable r) {
+            return doPrivileged(new PrivilegedAction<Thread>() {
+                public Thread run() {
+                    final Thread taskThread = new Thread(threadGroup, new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                r.run();
+                            } finally {
+                                msg.tracef("Thread [%s] exiting and executing cleanup tasks", Thread.currentThread().getName());
```

Comment:
Since `handleThreadExit` is already logging this information, this log message is not necessary.